### ISSUE TITLE
QM or HoP should get a printout of the Current Cargo Shuttle order from a fax machine/printer in their office.

### ISSUE THREAD

**Issue by Akbadain29** [unitystation/unitystation#4821]
## Description
Each Cargo Shuttle order text string inside CargoManger.cs is cleared when the Cargo Shuttle is sent back to Centcom for a new order. I think that when the Cargo Shuttle is arriving back with the new order (underneath where SpawnOrder() is called by CallShuttle()), a copy of the order should be sent to the QM's fax machine in their office and printed out on a piece of paper with a heading of "Order - 00X" and the contents. This would create a paper trail and help the QM/Security/Detective keep track of what is coming onto the station.

**Comment by Nukeuler123** [comment]
Would it be fine if the Supply Console itself printed the paper? That's how it is in the original.

**Comment by corp-0** [comment]
That sounds easily doable. Tho in the original they had 2 consoles, public one and the one behind the desk.

**Comment by Akbadain29** [comment]
Yeah, making the supply console print it out would be fine I think. It's probably important as that way the cargo techs can keep track of what was ordered too. The QM could then just ask for the printout if they need to file it away.

**Comment by Akbadain29** [comment]
It should just have the option in the inspector to allow the console to print out. That way if some people don't won't a particular console to print it out, they can easily disable it on the scene.

**Comment by Nukeuler123** [comment]
I was thinking of a checkbox to mark the console as "public" which would basically do the same thing but also enable/disable 
some functions

**Comment by corp-0** [comment]
It would require new UI to make it work like it did in the original

**Comment by Akbadain29** [comment]
If the "private" console already contains all the functions which are enabled, when you navigate tabs on the "public" console, maybe we should just show a padlock on the touchscreen instead of what the UI usually says. We already added the cargo id functionality to the console if I remember correctly, so it could be accompanied by an id required message.

**Comment by Nukeuler123** [comment]
I could see it where the public console shows a padlock on the shuttle control and cart screen but still allow people to be able to request items and have the computer automatically print a request paper.

**Comment by Akbadain29** [comment]
I'm a little bit confused though @Nukeuler123, is the request console only meant for mail orders not requesting items? I kept seeing an object called request console, which had a object placed in the map inside OutpostStation's cargo area.

**Comment by corp-0** [comment]
You had 2 consoles on cargo. The requests console is exposed to the public, you can fill a form there and it will print a request. You then go to the desk and give your printed request to the cargo tech who will approve/deny it and confirm it in their private console which is the one that actually buys and sends/retrieves the shuttle.

**Comment by Akbadain29** [comment]
Ok, in this issue I'm referring to Order receipts and not Approvals forms. I think that is clear, but I've written this comment just in case.

**Comment by corp-0** [comment]
that's true, but if we are gonna implement printable orders we might as well implement the entire thing. We would also close this ancient issue #2392

**Comment by Akbadain29** [comment]
I've been looking at a ss13 tutorial on youtube. They make it so that the crate has a piece of paper attached to the outside which you can tear off which contains the order, who requested it, the contents, required access to the crate itself. It's titled a Central Command Shipping Manifest. That probably is better then trying to print out at the console as it's physically attached to the order itself.

 Do you want me to ask a spriter to mock up a UI for the request console?

**Comment by Nukeuler123** [comment]
Well yes crates have those but most of the time there are two papers, one which is the request paper which has the requester's name and what they want and the second one which is on the box telling you the contents and who it goes to

**Comment by Akbadain29** [comment]
I think that the order should be just printed out at the supply console before we try anything too complicated and future-proof it.
So that way the order gets put into a List class, so it can be then further called/created/deleted between all the objects which need it.
